### PR TITLE
allow to use jargon file stored on module repo

### PR DESCRIPTION
**Pull request by bgoncalv**

### CODE REVIEW

File: check_modulemd.py [hunk 1]
```diff
@@ -66,23 +66,52 @@ def setUp(self):
...
-            else:
-                self.chkr = SpellChecker("en_US")
+                spell_dict = DictWithPWL(default_dict, jargonfile)
+            except:
+                self.error(
+                    "Could not initialize dictionary using %s file" % jargonfile)
+
+        if not spell_dict:
+            try:
+                spell_dict = DictWithPWL(default_dict)
```
Comment (mmathesius) [inline check_modulemd.py:90]: `DictWithPWL()` should always be called with at least two arguments. Either explicitly specify the second argument as `None`, or call `Dict` instead.

File: check_modulemd.py [hunk 2]
```diff
@@ -66,23 +66,52 @@ def setUp(self):
...
+        #add words from module name as jargon
+        for w in self.mmd.name.split('-'):
+            spell_dict.add_to_session(w)
+
+        try:
+            chkr = SpellChecker(spell_dict)
         except:
             self.error(
-                "Could not initialize spell checker with dictionary %s" % dict)
+                "Could not initialize spell checker with dictionary %s" % spell_dict)
```
Comment (mmathesius) [inline check_modulemd.py]: Printing an enchant dictionary object as a string isn't going to provide a useful message.

File: check_modulemd.py (hunk 1 above)
Comment (bgoncalv) [reply check_modulemd.py:90]: I thought by default it use None for the other arguments:
class enchant.DictWithPWL(tag, pwl=None, pel=None, broker=None), but sure I can change it.

Comment (mmathesius) [reply check_modulemd.py:90]: Apologies. No need to update. DictWithPWL() is new to me and I misread the spec. But every example I could find of its usage always included two or more arguments--with the second arg possibly being None.

**Review by mmathesius — APPROVED** [review]

**Pull request merged**